src/main/groovy/com/calgaryscientific/gradle/VeracodeGetBuildListTask.groovy

```diff
@@ -26,18 +26,23 @@
 
 package com.calgaryscientific.gradle
 
+import groovy.transform.CompileStatic
+
+@CompileStatic
 class VeracodeGetBuildListTask extends VeracodeTask {
     static final String NAME = 'veracodeGetBuildList'
+    private String app_id
 
     VeracodeGetBuildListTask() {
         description = "List builds for the given 'app_id'"
         requiredArguments << 'app_id'
+        app_id = project.findProperty("app_id")
+        defaultOutputFile = new File("${project.buildDir}/veracode", "build-list-${app_id}.xml")
     }
 
     void run() {
-        xmlio.writeXml('build-list.xml', uploadAPI().getBuildList(project.app_id)
-        ).each() { build ->
-            println "${build.@build_id}=${build.@version}"
-        }
+        Node xml = XMLIO.writeXml(getOutputFile(), veracodeAPI.getBuildList(app_id))
```

Comment:
again, do we have an example getBuildList(app_id) would return in test?

Reply:
The test file shows the output. I think the tasks do too little to care for a test that is why I didn't add one. I am in the middle of something else but didn't want to mix this with the rest of the changes.